ledger/eval/eval_test.go

```diff
@@ -403,9 +402,11 @@ int 1`,
 					},
 				}
 
-			var expectedEvents []mocktracer.Event
+			var expectedEvents = []mocktracer.Event{mocktracer.BeforeBlock(eval.block.Round())}
 			if testCase.firstTxnBehavior == "approve" {
-				expectedEvents = mocktracer.FlattenEvents([][]mocktracer.Event{
+				err = eval.endOfBlock()
```

Comment:
The tests, as is, wouldn't ever call `endOfBlock` since they're just exercising the logic evaluation and not the full block lifecycle, so I added this here. I did think that the only place where evaluation errors were being created was after the first txn so I assumed that all others would pass. 

If we don't need all of the txn evals to pass in order to successfully call `endOfBlock` then I guess it's still a fine test to simulate the tracer as though the full eval had passed.